### PR TITLE
[DO NOT MERGE] AUTO inferences with CPU while the actual accelerator loads the network (CVS-56054)	

### PR DESCRIPTION
**Pull request by myshevts**
This is really a draft and lot's of semantic/behaviour opens to address and/or raise to the Arch forum, e.g.

So the TODOs are (marked as "fixme in the code)":
- shall the SetConfig/GetConfig keep the settings and reapply to the actual (accelerator)  network when it is ready.
- what the GetRemoteContext should return (e.g. shall it wait for the actual accelerator to load the network first?)
- Metrics etc
- finally, there are subtle details of asynchronocity (e.g. calling the StartAsync from two threads simultaneously <which should normally throw an exception), while the devices HotSwap amy happen in-between). Currently the implemented solution is likely incomplete.
All these should be verified with (updated) behaviour tests.

### CODE REVIEW

File: inference-engine/src/auto_plugin/auto_infer_request.cpp [hunk 1]
```diff
@@ -11,32 +12,41 @@ namespace AutoPlugin {
...
-    return _inferRequest->GetBlob(name);
+    return IInferRequestInternal::GetBlob(name);
 }
 
 void AutoInferRequest::Cancel() {
     _inferRequest->Cancel();
 }
 
 void AutoInferRequest::StartAsync() {
+    HotSwapRequests(); //safe to call here (before actual inference started)
```
Comment (myshevts) [inline inference-engine/src/auto_plugin/auto_infer_request.cpp:48]: this is a caveat on async: as calling the STartAsync from second therad should throw (REQUEST_BUSY, afaik), we need to make sure that another StartAsync still would throw even though the HotSwap has happend in the "current" StartAsync.

Maybe creating a special callback (that alos wraps original user's callback is better idea).
Or, inheriting the AutoRequest from the InferRequestThreadSafeDefault and checking it's state (as opposite to relying on the BUSY state of the device's request) is also possible

File: inference-engine/src/auto_plugin/auto_infer_request.cpp [hunk 2]
```diff
@@ -11,32 +12,41 @@ namespace AutoPlugin {
...
+    for (const auto &it : _networkOutputs)
+        _outputs[it.first] = _inferRequest->GetBlob(it.first);
 }
 
 std::map<std::string, InferenceEngine::InferenceEngineProfileInfo> AutoInferRequest::GetPerformanceCounts() const {
     return _inferRequest->GetPerformanceCounts();
 }
 
 void AutoInferRequest::InferImpl() {
+    HotSwapRequests(); //safe to call here (before actual inference started)
```
Comment (myshevts) [inline inference-engine/src/auto_plugin/auto_infer_request.cpp:30]: Pls see a similar comment on the StartAsync

**Review by mashoujiang — COMMENTED** [review]
I would like change the code based on my understanding as the comments if you think that's right.

**Comment by myshevts** [comment]
brushed and merged as #5944

**Pull request closed without merging**